### PR TITLE
Improve Public Interface Types

### PR DESCRIPTION
**Pull request by max-muoto**
Some minor improvements to the public interface types. The main QoL improvement is probably going to be fixing type-checking when passing in unioned/intersected `pgtrigger.Q` instances.

### CODE REVIEW

File: pgtrigger/core.py [hunk 1]
```diff
@@ -187,9 +190,9 @@ class Timing(_Primitive):
 class Condition:
     """For specifying free-form SQL in the condition of a trigger."""
 
-    sql: str = None
+    sql: str | None = None
```
Comment (wesleykendall) [inline pgtrigger/core.py:193]: I'm assuming you'll need to use `Union` for 3.8 compat, right?

Comment (max-muoto) [reply pgtrigger/core.py:193]: Nope, since I added `from __future__ import annotations`

Comment (max-muoto) [reply pgtrigger/core.py:193]: You can use it in earlier versions when type annotations are stringified.

Comment (wesleykendall) [reply pgtrigger/core.py:193]: Oh nice, I should convert this elsewhere. I do believe there is a py3.8 bug if you check circle though

```    from pgtrigger.runtime import (
  File "/code/pgtrigger/runtime.py", line 154, in <module>
    def ignore(*uris: str, databases: Union[List[str], None] = None) -> Generator[None]:
TypeError: 'ABCMeta' object is not subscriptable```

**Review by wesleykendall — APPROVED** [review]
looks good once tests pass

File: pgtrigger/core.py (hunk 1 above)
Comment (max-muoto) [reply pgtrigger/core.py:193]: Ah, had to add a future imports for that.

**Pull request merged**